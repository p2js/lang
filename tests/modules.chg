mod a {
    mod b {
        fn c(): i32 { return 1; }
    }
    fn d(): i32 { return 2; }
    type E = { v: i32 };
}
fn main(): i32 {
    return a::d() + a::b::c();
}
