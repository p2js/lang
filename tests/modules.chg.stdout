expr @ 3:30 3:31 => i32
expr @ 3:23 3:31 => never
expr @ 3:21 4:0 => ()
expr @ 5:26 5:27 => i32
expr @ 5:19 5:27 => never
expr @ 5:17 6:0 => ()
expr @ 9:15 9:16 => fn d(...)
expr @ 9:15 9:18 => i32
expr @ 9:27 9:28 => fn c(...)
expr @ 9:27 9:30 => i32
expr @ 9:15 9:30 => i32
expr @ 9:5 9:30 => never
expr @ 8:16 11:1 => ()



define i32 @root$a$d() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 2, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}

define i32 @root$a$b$c() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
store i32 1, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}

define i32 @main() {
prologue:
%l.0 = alloca i32
br label %bb.0

bb.0:
%l.1 = call i32 @root$a$d()
br label %bb.1

bb.1:
%l.2 = call i32 @root$a$b$c()
br label %bb.2

bb.2:
%l.3 = add i32 %l.1, %l.2
store i32 %l.3, i32* %l.0
%ret.0 = load i32, i32* %l.0
ret i32 %ret.0

}


